std/exception.d

```diff
@@ -2145,3 +2145,58 @@ pure nothrow @safe unittest
 
     auto infSlice = infinite[0 .. $]; // this would throw otherwise
 }
+
+
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
```

Comment:
I mean that `enforce` is designed to require a constructor that takes the `msg`, `file`, and `line` number - and in that order - which is what this template mixin gives you. So, by using this template mixin, your exception will work with `enforce`, whereas if you use a non-standard set of parameters for your exception's constructor (or a non-standard order), then it won't work with `enforce`. So, changing the end of that sentence to be "`exception was thrown from or to make it work with $(LREF enforce)`" points out that part of the reason to use this `mixin` is to ensure that your exception type works with `enforce`.